packages/devtools_app/lib/src/debugger/flutter/breakpoints.dart

```diff
@@ -5,55 +5,103 @@
 
-class BreakpointPicker extends StatelessWidget {
-  const BreakpointPicker({Key key, @required this.controller})
-      : super(key: key);
+class BreakpointPicker extends StatefulWidget {
+  const BreakpointPicker({
+    Key key,
+    @required this.controller,
+    @required this.selected,
+    @required this.onSelected,
+  }) : super(key: key);
 
   final DebuggerController controller;
+  final BreakpointAndSourcePosition selected;
+  final void Function(BreakpointAndSourcePosition breakpoint) onSelected;
 
-  String textFor(Breakpoint breakpoint) {
-    if (breakpoint.resolved) {
-      final location = breakpoint.location as SourceLocation;
-      // TODO(djshuckerow): Resolve the scripts in the background and
-      // switch from token position to line numbers.
-      return '${location.script.uri.split('/').last} Position '
-          '${location.tokenPos} (${location.script.uri})';
-    } else {
-      final location = breakpoint.location as UnresolvedSourceLocation;
-      return '${location.script.uri.split('/').last} Position '
-          '${location.line} (${location.script.uri})';
-    }
-  }
+  @override
+  _BreakpointPickerState createState() => _BreakpointPickerState();
+}
 
+class _BreakpointPickerState extends State<BreakpointPicker> {
   @override
   Widget build(BuildContext context) {
-    return densePadding(
-      Scrollbar(
-        child: SingleChildScrollView(
-          scrollDirection: Axis.horizontal,
-          child: SizedBox(
-            width: MediaQuery.of(context).size.width,
-            child: ValueListenableBuilder(
-              valueListenable: controller.breakpoints,
-              builder: (context, breakpoints, _) {
-                return ListView.builder(
-                  itemCount: breakpoints.length,
-                  itemExtent: defaultListItemHeight,
-                  itemBuilder: (context, index) => SizedBox(
-                    height: CodeView.rowHeight,
-                    child: Text(textFor(breakpoints[index])),
-                  ),
-                );
-              },
-            ),
+    return ValueListenableBuilder<List<BreakpointAndSourcePosition>>(
+      valueListenable: widget.controller.breakpointsWithLocation,
+      builder: (context, breakpoints, _) {
+        return ListView.builder(
+          itemCount: breakpoints.length,
+          itemExtent: defaultListItemHeight,
+          itemBuilder: (context, index) {
+            return buildBreakpoint(context, breakpoints[index]);
+          },
+        );
+      },
+    );
+  }
+
+  Widget buildBreakpoint(BuildContext context, BreakpointAndSourcePosition bp) {
+    final subtleStyle =
+        TextStyle(color: Theme.of(context).unselectedWidgetColor);
+
+    return Material(
+      color: bp.id == widget.selected?.id
+          ? Theme.of(context).selectedRowColor
+          : null,
+      child: InkWell(
+        onTap: () => widget.onSelected(bp),
+        child: densePadding(
+          Row(
```

Comment:
I'm having some overflow issues here - when the width for the breakpoint view gets too narrow, we see an overflow in this Row.

Reply:
<img width="314" alt="Screen Shot 2020-04-16 at 8 40 30 AM" src="https://user-images.githubusercontent.com/1269969/79476534-f4278a80-7fbd-11ea-9154-d320e4a898ed.png">

Reply:
(this has been resolved)